### PR TITLE
added descriptions to all configuration parameters

### PR DESCRIPTION
**Pull request by Byron**
This is the first of many more PRs. They are aimed to make this plugin as easy-to-use as possible. Personally I was unable to get rust-completion to work with this plugin, which is my main motivation for contributing to it.

In order to tackle this, I would write plenty of unit-tests, integration-tests and functional-tests to verify the existing code and protect from regressions. Please let me know if you have any suggestions on how you see this :) !

### CODE REVIEW

File: lib/config.coffee [hunk 1]
```diff
@@ -3,19 +3,28 @@ module.exports =
...
     order: 4
+    description: 'Additional configuration for ycmd. Follow
+      [-> this link <-](https://github.com/Valloric/YouCompleteMe#the-gycm_global_ycm_extra_conf-option)
+      for more information.'
   rustSrcPath:
     type: 'string'
     default: ''
     order: 5
+    description: 'The directory containing the Rust source code, e.g. `/path/to/rust-lang/rust/src`.
+      You have to to add the `rs` file extension in **Enabled Filetypes** and restart atom
```
Comment (Qusic) [inline lib/config.coffee]: It seems rust language support hasnt been included in atom release.
If used with the package `language-rust `, the filetype to add is `rust`. https://github.com/zargony/atom-language-rust/blob/master/settings/rust.cson#L1

File: lib/config.coffee [hunk 2]
```diff
@@ -3,19 +3,28 @@ module.exports =
...
   ycmdPath:
     type: 'string'
     default: '/usr/lib/youcompleteme/third_party/ycmd'
     order: 2
+    description: 'The directory containing the `ycmd/default_settings.json` file.'
   enabledFiletypes:
     type: 'string'
     default: 'c, cpp, objc, objcpp'
     order: 3
+    description: 'A comma-separated list of file-extensions within we should provide suggestions.'
```
Comment (Qusic) [inline lib/config.coffee]: they are not file extensions. they are the language components of the editor's root scope descriptor. atom doesn't have much documentation on this yet. https://github.com/atom/atom/issues/8430
here is a table for reference(see entries like `.source.js`): https://github.com/atom/autocomplete-plus/wiki/Autocomplete-Providers

**Comment by Qusic** [comment]
> You should strive to be so clear in your naming of the setting that you do not need to specify a title or description!
https://atom.io/docs/api/v1.4.0/Config#title-and-description

Actually I think these config names are quite clear for YouCompleteMe users and adding some links to related documentation should be enough.

**Comment by Byron** [comment]
Thanks for your input - I will make the changes and push them in another commit.
Do you think that changes in the configuration will automatically be picked up ?
I wasn't sure about that when skimming over the latest code.

**Comment by kkHAIKE** [comment]
Good work!!

**Comment by Qusic** [comment]
Yes changes will take effect without restarting atom now.

**Comment by Byron** [comment]
Great to hear. Also, I have updated the descriptions.
While writing the part about Rust, I thought it might actually be worth to auto-enable it if there is a rust source directory. What do you think about that ?
Besides, if you like to see other/different descriptions, please feel free to modify the ones provided here - it might be easiest after all.

**Comment by Qusic** [comment]
I am afraid auto-enabling will make the settings confusing. After all the default value is not dynamic.

**Comment by Byron** [comment]
After re-reading Valloric/ycmd#121, I also came to believe that it's worth waiting. Those who seek Snippet support for Rust can always use [atom-racer](https://github.com/edubkendo/atom-racer) in the meanwhile, which seemed to work reasonably well for me.

**Pull request merged**